### PR TITLE
updates to the latest version of R, and changes the RNG to use provided C function pointers

### PR DESCRIPTION
**Pull request by simonbyrne**
Also makes changes to Makefile to make updates easier in future.

Replaces #13.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: c
+os:
+  - linux
+  - osx
+notifications:
+  email: false
+script:
+  - make -j3
```
Comment (tkelman) [inline .travis.yml]: are there any tests?

Comment (simonbyrne) [inline .travis.yml]: The tests are all in Rmath.jl

**Comment by tkelman** [comment]
Guess we have to keep a close eye on whether R ever deletes any of these files?

**Comment by simonbyrne** [comment]
True, we won't catch deletions, unless I delete all the files before re-extracting them.

**Comment by simonbyrne** [comment]
I should also explain that the only files which aren't from R in `src/` are `callback.c` and the `Makefile`.

**Comment by ararslan** [comment]
Bump. Is there anything that's holding this up? I'd be nice to add deployment to the Travis YAML once this is merged so that we can include binaries in the releases.

File: src/callback.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Function pointers for callbacks for random number generators.
+ */
+double (*unif_rand_ptr)(void) = 0;
+double (*norm_rand_ptr)(void) = 0;
+double (*exp_rand_ptr)(void) = 0;
+
+double unif_rand(void)
+{
+    return (*unif_rand_ptr)();
+}
```
Comment (simonbyrne) [inline src/callback.c]: My only remaining question is if it would be better if this were a header-defined `inline` function?

Comment (simonbyrne) [inline src/callback.c]: I guess this can be changed later without breaking anything.

**Pull request merged**